### PR TITLE
Added model configuration

### PR DESCRIPTION
**Pull request by DenKorporation**
The following has been added:

- EntityTypeConfiguration for all models

- DbContext

- Db Seeder 

- Extension method for service provider

- Docker support for postgreSQL

### CODE REVIEW

File: src/EShop.DAL/Context/Configurations/CategoryEntityTypeConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public class CategoryEntityTypeConfiguration : IEntityTypeConfiguration<Category>
+{
+    public void Configure(EntityTypeBuilder<Category> builder)
+    {
+        builder
+            .HasKey(c => c.Id);
+
+        builder
+            .Property(c => c.Id)
+            .ValueGeneratedOnAdd()
```
Comment (SUPERKISA8BIIT) [inline src/EShop.DAL/Context/Configurations/CategoryEntityTypeConfiguration.cs]: this is redundant. check everywhere

File: src/EShop.DAL/Context/Configurations/CategoryEntityTypeConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            .IsRequired();
```
Comment (SUPERKISA8BIIT) [inline src/EShop.DAL/Context/Configurations/CategoryEntityTypeConfiguration.cs]: this is redundant. id is required by default. check everywhere

File: src/EShop.DAL/Context/Configurations/CategoryEntityTypeConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+        builder
+            .Property(c => c.Name)
+            .HasMaxLength(100)
+            .IsRequired();
+
+        builder
+            .HasIndex(c => c.Name)
+            .IsUnique();
+    }
```
Comment (SUPERKISA8BIIT) [inline src/EShop.DAL/Context/Configurations/CategoryEntityTypeConfiguration.cs:27]: where is configuration of relation with product table

Comment (DenKorporation) [reply src/EShop.DAL/Context/Configurations/CategoryEntityTypeConfiguration.cs:27]: ProductEntityTypeConfiguration.cs

File: src/EShop.DAL/Context/EShopContext.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace EShop.DAL.Context;
+
+public class EShopContext : DbContext
+{
+    public EShopContext(DbContextOptions<EShopContext> options) : base(options)
+    {
+    }
+
+    public DbSet<Category> Categories { get; set; }
```
Comment (SUPERKISA8BIIT) [inline src/EShop.DAL/Context/EShopContext.cs:13]: it's better to write properties before the constructor

**Review by SUPERKISA8BIIT — COMMENTED** [review]
checked

**Pull request merged**